zebra-state/src/service.rs

```diff
@@ -333,20 +324,26 @@ impl StateService {
             tokio::sync::mpsc::unbounded_channel();
 
         let finalized_state_for_writing = finalized_state.clone();
-        let chain_tip_sender_for_writing = chain_tip_sender.clone();
+        let non_finalized_state_for_writing = non_finalized_state.clone();
```

Comment:
Required Fix: Cloning creates an independent clone of the non-finalized state, so `StateService.mem` never gets updated. Using an empty non-finalized state might cause some test failures.

Instead, remove `StateService.mem`, and use `StateService.read_service.non_finalized_state_receiver` to get a cached version of the latest non-finalized state.

`StateService.disk` is also the same as `StateService.read_service.db`, so let's remove `StateService.disk` at the same time.

Reply:
Oops. I removed `mem` and `disk` in [`64f1004`](https://github.com/ZcashFoundation/zebra/pull/5257/commits/64f1004d470c9455a71680f0167097daddcd8949) and the three commits that followed that one.

Reply:
Thanks!

I wanted to get rid of the `mem` name, because it's a bit confusing, so I renamed it in 8f9aea5 in PR #5299.